src/omniperf_profile/profiler_base.py

```diff
@@ -49,7 +245,10 @@ def pre_processing(self):
         """Perform any pre-processing steps prior to profiling.
         """
         logging.debug("[profiling] pre-processing using %s profiler" % self.__profiler)
-
+        
+        # verify soc compatibility
+        if self.__profiler not in self._soc.get_compatible_profilers():
+            error("%s is not enabled in %s. Available profilers include: %s" % (self._soc.get_soc_name(), self.__profiler, self._soc.get_compatible_profilers()))
```

Comment:
I like the printout of the compatible profilers in this error message. Maybe consider doing similar for the related message in src/omniperf_base.py? (L138).